### PR TITLE
more acceptance tests

### PR DESCRIPTION
**Pull request by hedwigz**

### CODE REVIEW

File: atlas/resource_atlas_schema_test.go [hunk 1]
```diff
@@ -106,3 +113,154 @@ func TestAccAtlasDatabase(t *testing.T) {
...
+	schema "main" {
+	}
+	EOT
+  url = "sqlite://database.sqlite3?cache=shared"
+}
+`
+
+func TestAccInvalidSchemaReturnsError(t *testing.T) {
+	defer func() {
+		os.Remove("database.sqlite3")
```
Comment (rotemtam) [inline atlas/resource_atlas_schema_test.go:153]: use t.Cleanup

File: atlas/resource_atlas_schema_test.go [hunk 2]
```diff
@@ -11,9 +13,14 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
-const testAccActionConfigCreate = `
+const (
+	MYSQL_URL     = "mysql://root:pass@localhost:3306/test"
+	MYSQL_DEV_URL = "mysql://root:pass@localhost:3307/test"
```
Comment (a8m) [inline atlas/resource_atlas_schema_test.go:18]: Keep it private.

File: atlas/resource_atlas_schema_test.go [hunk 3]
```diff
@@ -106,3 +113,154 @@ func TestAccAtlasDatabase(t *testing.T) {
...
+		charset = "utf8mb4"
+		collate = "utf8mb4_0900_ai_ci"
+	}
+	EOT
+}
+resource "atlas_schema" "testdb" {
+  hcl = data.atlas_schema.sanity.hcl
+  url = "%s"
+}
+`, MYSQL_DEV_URL, MYSQL_URL)
```
Comment (a8m) [inline atlas/resource_atlas_schema_test.go:225]: why it's outside the function?

File: atlas/resource_atlas_schema_test.go [hunk 4]
```diff
@@ -106,3 +113,154 @@ func TestAccAtlasDatabase(t *testing.T) {
...
+
+const sanityState = `table "type_table" {
```
Comment (a8m) [inline atlas/resource_atlas_schema_test.go:227]: ditto

File: atlas/resource_atlas_schema_test.go [hunk 5]
```diff
@@ -106,3 +113,154 @@ func TestAccAtlasDatabase(t *testing.T) {
...
+		t.Error(fmt.Errorf("expected database to have table \"orders\" but got: %s", realm.Schemas[0].Tables[0].Name))
+		return
+	}
+}
+
+const createTableStmt = `create table type_table
+(
+    tBit                        bit(10)              default 4                                              null,
+    tInt                        int(10)              default 4                                               not null,
+    tTinyInt                    tinyint(10)          default 8                                                   null
```
Comment (a8m) [inline atlas/resource_atlas_schema_test.go:198]: please fix the formatting

File: atlas/resource_atlas_schema_test.go [hunk 6]
```diff
@@ -106,3 +113,154 @@ func TestAccAtlasDatabase(t *testing.T) {
...
+    default = 4
+  }
+}
+schema "test" {
+  charset = "utf8mb4"
+  collate = "utf8mb4_0900_ai_ci"
+}
+`
+
+func TestAccRemoveColumns(t *testing.T) {
```
Comment (a8m) [inline atlas/resource_atlas_schema_test.go:241]: What is "Acc"?

File: atlas/resource_atlas_schema_test.go [hunk 7]
```diff
@@ -106,3 +113,154 @@ func TestAccAtlasDatabase(t *testing.T) {
...
+		}
+	}
+	schema "main" {
+	}
+	EOT
+  url = "sqlite://database.sqlite3?cache=shared"
+}
+`
+
+const testAccInvalidSqliteSchema = `
```
Comment (a8m) [inline atlas/resource_atlas_schema_test.go:134]: why it's global?
```suggestion
const testAccInvalidSQLiteSchema = `
```

Comment (a8m) [reply atlas/resource_atlas_schema_test.go:134]: Also, why it's invalid?

File: atlas/resource_atlas_schema_test.go (hunk 6 above)
Comment (hedwigz) [reply atlas/resource_atlas_schema_test.go:241]: https://www.terraform.io/plugin/sdkv2/testing/acceptance-tests#test-files

File: atlas/resource_atlas_schema_test.go (hunk 7 above)
Comment (hedwigz) [reply atlas/resource_atlas_schema_test.go:134]: added comment

File: atlas/resource_atlas_schema_test.go [hunk 8]
```diff
@@ -11,16 +12,22 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
-const testAccActionConfigCreate = `
+const (
+	mysql_url     = "mysql://root:pass@localhost:3306"
```
Comment (rotemtam) [inline atlas/resource_atlas_schema_test.go:16]: go vars usually `camelCased`, no?

File: atlas/resource_atlas_schema_test.go [hunk 9]
```diff
@@ -107,30 +112,199 @@ func TestAccAtlasDatabase(t *testing.T) {
...
+				Check: func(s *terraform.State) error {
+					cli, err := sqlclient.Open(context.Background(), mysql_url)
+					if err != nil {
+						return err
+					}
+					realm, err := cli.InspectRealm(context.Background(), nil)
+					if err != nil {
+						return err
+					}
+
```
Comment (rotemtam) [inline atlas/resource_atlas_schema_test.go:187]:
```suggestion
```

**Pull request merged**